src/gentropy/dataset/variant_index.py

```diff
@@ -51,17 +146,20 @@ def add_annotation(
         # Columns in the source dataset:
         variant_index_columns = [
             # Combining cross-references:
-            f.array_union(f.col("dbXrefs"), f.col("annotation_dbXrefs"))
+            self._merge_columns(f.col("dbXrefs"), f.col("annotation_dbXrefs")).alias(
```

Comment:
To avoid code duplication, what i suggested earlier is to define in the function all columns that can be extended and then iterate over them, so you only reference the `merge` method in the iterable. Something like
```
    # Rename columns in the annotation source to avoid conflicts
    ...

    # Columns to be extended
    all_extendable_cols = {
        "dbXrefs": "annotation_dbXrefs",
        "inSilicoPredictors": "annotation_inSilicoPredictors",
        "alleleFrequencies": "annotation_alleleFrequencies"
    }

    # Columns in the source dataset
    variant_index_columns = []
    for col in self.df.columns:
        if col in all_extendable_cols:
            extended_col = self._merge_columns(f.col(col), f.col(all_extendable_cols[col])).alias(col)
            variant_index_columns.append(extended_col)
        else:
            variant_index_columns.append(f.col(col))

    # Join the annotation to the dataset
    ...
```

Reply:
This piece of code is completely re-written. The actual logic ready for the following things:

- If a column only coming from the annotation it will be picked up and propagated to the resulting index.
- If the column can be found in both, depending on the type of the column, might safe union is taken (if array) otherwise we coalesce.